### ISSUE TITLE
Passing self by raw pointer

### ISSUE THREAD

**Issue by mahkoh** [rust-lang/rfcs#1460]
This seems like a useful feature especially for traits. Why isn't this allowed?

**Comment by bluss** [comment]
It was removed (while entirely buggy) in https://github.com/rust-lang/rust/pull/26225

**Comment by mahkoh** [comment]
If it was removed without an RFC, then it should be possible to fix the underlying problems and re-enable them without an RFC too.

**Comment by bluss** [comment]
Not really. It was never an intentional feature. I can't find an issue or rfc for generalized Self in my searching at the moment, but you might look for that.

**Comment by mahkoh** [comment]
>I can't find an issue or rfc for generalized Self

Most of the language wasn't introduced via an RFC.

>It was never an intentional feature

From the PR you posted, it seems that it was explicitly allowed before it was removed: https://github.com/rust-lang/rust/pull/26225/files#diff-4ed25c00aceb84666fca639cf8101c7cL2177 How can this be unintentional?

>I can't find an issue or rfc for generalized Self in my searching at the moment, but you might look for that.

Yes, generalized Self is one of the reasons that this should be allowed. It's one of the things that forces Box to be special cased.

**Comment by mahkoh** [comment]
In the absence of of `&own`, raw pointers are the only way to pass traits by value.

**Comment by arielb1** [comment]
It didn't actually work before it was removed - it was another syntax for by-value `Self`.
